### PR TITLE
Add support for Synergy v2 and v3, and for specifying edition

### PR DESCRIPTION
**Pull request by giorgionegro**
added support  for the new "version" of the key

### CODE REVIEW

**Comment by Richienb** [comment]
Readme needs to be updated with the same info as in the typings

**Comment by giorgionegro** [comment]
> Readme needs to be updated with the same info as in the typings

should be good now

**Comment by giorgionegro** [comment]
maybe version 2 should be the default since it should work with both synergy 1 and 3

**Comment by Richienb** [comment]
> maybe version 2 should be the default since it should work with both synergy 1 and 3

Does this mean v1 works on Synergy 1 and v2 works on Synergy 1, 2 and 3?

Docs would need to be updated then.

**Comment by giorgionegro** [comment]
> > maybe version 2 should be the default since it should work with both synergy 1 and 3
> 
> Does this mean v1 works on Synergy 1 and v2 works on Synergy 1, 2 and 3?
> 
> Docs would need to be updated then.

yes, v2 serial works in all three. just checked with 1.14

**Comment by Richienb** [comment]
Because it only works on recent versions of v1, it would make it a breaking change.
The default should still be 1, just to be safe.

**Comment by giorgionegro** [comment]
> Because it only works on recent versions of v1, it would make it a breaking change. The default should still be 1, just to be safe.

now it should be okay

File: index.js [hunk 1]
```diff
@@ -5,12 +5,21 @@ const bufferFrom = require("buffer-from")
...
+
+    if (version === 1) {
+        return toHex(`{v1;${edition};${name};${userLimit};${email};${business};0;0}`)
+    }
 
-    const key = `{v1;pro;${name};${userLimit};${email};${business};0;0}`
-    return bufferFrom(key).toString("hex").toUpperCase()
+    if (version === 2) {
+        return toHex(`{v2;maintenance;${edition};${name};${userLimit};${email};${business};0;0}`)
+    }
```
Comment (Richienb) [inline index.js:24]: At the end here, throw an error if an invalid version was provided. ``Expected version to be `1` or `2`, got ${version}``

**Pull request merged**